### PR TITLE
feat: expose `derived_address` helper on `Descriptor`

### PR DESCRIPTION
**Pull request by reez**
### Description

exposing a helper to derive an address from a descriptor without building a wallet

follows miniscript’s recommended at_derivation_index → derived_descriptor flow to avoid panicy edge cases with unsupported descriptor https://docs.rs/miniscript/12.3.5/src/miniscript/descriptor/mod.rs.html#886

### Notes to the reviewers

tests covering both the happy path and multipath error mapping

### Documentation

The helper includes these

- [x] [`bdk_wallet`](https://docs.rs/bdk_wallet/latest/bdk_wallet/)

  https://docs.rs/bdk_wallet/latest/bdk_wallet/descriptor/enum.Descriptor.html#method.at_derivation_index
  https://docs.rs/bdk_wallet/latest/bdk_wallet/descriptor/enum.Descriptor.html#method.derived_descriptor-1
  
- [x] [`bitcoin`](https://docs.rs/bitcoin/latest/bitcoin/index.html)

  https://docs.rs/bitcoin/0.32.7/bitcoin/address/struct.Address.html#method.from_script

- [ ] [`uniffi`](https://github.com/mozilla/uniffi-rs) 
- [ ] Other: 

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing
* [ ] I've added a changelog in the next release tracking issue (see [example](https://github.com/bitcoindevkit/bdk-ffi/issues/875))
* [x] I've linked the relevant upstream docs or specs above

#### New Features:

* [x] I've added tests for the new feature
* [ ] I've added docs for the new feature

#### Bugfixes:

* [ ] This pull request breaks the existing API
* [ ] I've added tests to reproduce the issue which are now passing
* [ ] I'm linking the issue being fixed by this PR

### CODE REVIEW

**Comment by reez** [comment]
CI passes except for Android but I think that's a known thing atm

**Review by thunderbiscuit — COMMENTED** [review]
I'm not sure of the exact difference between the two approaches (also the bdk_wallet type is a wrapper around the rust-bitcoin type and so has new methods I wasn't aware of initially), so let me know if this is wrong somehow but I was able to do away with the use of `dervied_descriptor` entirely and simplify quite a bit (mind you with handling the errors it will get bigger):

Currently:
```rust
let secp = Secp256k1::verification_only();
let derived_descriptor = self
    .extended_descriptor
    .at_derivation_index(index)
    .and_then(|desc| desc.derived_descriptor(&secp))
    .map_err(|error| match error {
        ConversionError::HardenedChild => DescriptorError::HardenedDerivationXpub,
        ConversionError::MultiKey => DescriptorError::MultiPath,
    })?;

let script_pubkey = derived_descriptor.script_pubkey();
let address =
    Address::from_script(Arc::new(Script(script_pubkey)), network).map_err(|error| {
        DescriptorError::Miniscript {
            error_message: error.to_string(),
        }
    })?;

Ok(Arc::new(address)
```

Can be rewritten as:

```rust
let definite_descriptor = self.extended_descriptor.at_derivation_index(index).unwrap();
let address = definite_descriptor.address(network).unwrap();
        
Ok(Arc::new(address.into()))
```

Let me know if I'm missing something.

File: bdk-ffi/src/descriptor.rs [hunk 1]
```diff
@@ -355,6 +357,33 @@ impl Descriptor {
     pub fn desc_type(&self) -> DescriptorType {
         self.extended_descriptor.desc_type()
     }
+
+    pub fn derived_address(
```
Comment (thunderbiscuit) [inline bdk-ffi/src/descriptor.rs]: This method doesn't have a direct 1-to-1 match on the Rust side, but I was just thinking about how this is using a past tense/passive naming style vs the active `Descriptor::derive_address`. I think the convention is more on the active style, but this is a nit and not a blocker. Any opinion on this @reez?

Comment (reez) [reply bdk-ffi/src/descriptor.rs]: small nit but important one, totally agree, you're right. let me change it right now.

Comment (reez) [reply bdk-ffi/src/descriptor.rs]: updated now, good review/catch 👍

**Review by thunderbiscuit — APPROVED** [review]
ACK f21ac743c2ef423cb4ce11ae3974e3c0436eaea5
